packages/react/src/components/Code/index.tsx

```diff
@@ -23,25 +23,32 @@ interface Props extends SyntaxHighlighterProps {
   language?: 'javascript' | 'css' | 'html' | 'yaml';
   className?: string;
   tabIndex?: number;
+  ariaLabel?: string;
 }
 
 const Code: React.ComponentType<React.PropsWithChildren<Props>> = ({
   children,
   className,
   tabIndex,
+  ariaLabel,
   ...props
-}) => (
-  <>
-    <Highlighter
-      {...props}
-      useInlineStyles={false}
-      className={classNames('Code', className)}
-      tabIndex={tabIndex}
-    >
-      {children}
-    </Highlighter>
-  </>
-);
+}) => {
+  if (!ariaLabel) ariaLabel = 'Code snippet';
+
+  return (
+    <>
+      <Highlighter
+        {...props}
+        useInlineStyles={false}
+        className={classNames('Code', className)}
+        tabIndex={tabIndex}
+        {...(tabIndex === 0 && { role: 'region', 'aria-label': ariaLabel })}
```

Comment:
This could cause problems in cases where the accessible name for the code block is added via `aria-labelledby`

```js
<h3 id="foo">Some heading</h3>
<Code aria-labelledby="foo">some code</Code> // cauldron adds an `aria-label` which will override/conflict with the aria-labelledby value
```

I don't think cauldron should enforce how an accessible name is added to a code block

Reply:
Good suggestion! Updated.